src/canvas/StaticCanvas.ts

```diff
@@ -45,10 +45,19 @@ import { staticCanvasDefaults } from './StaticCanvasOptions';
 import { log, FabricError } from '../util/internals/console';
 import { getDevicePixelRatio } from '../env';
 
-export type TCanvasSizeOptions = {
-  backstoreOnly?: boolean;
-  cssOnly?: boolean;
-};
+/**
+ * Having both options in TCanvasSizeOptions set to true transform the call in a calcOffset
+ * Better try to restrict with types to avoid confusion.
+ */
+export type TCanvasSizeOptions =
+  | {
+      backstoreOnly?: true;
+      cssOnly?: false;
+    }
+  | {
+      backstoreOnly?: false;
+      cssOnly?: true;
```

Comment:
```suggestion
  | {
      backstoreOnly?: true;
      cssOnly?: never;
    }
  | {
      backstoreOnly?: never;
      cssOnly?: true;
```

should be never or undefined, not sure

Reply:
i tried this, it gets angry.  false is still a valid option you could pass, the important is that you don't do the double true.